runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -915,6 +915,16 @@ private Debuggee registerDebuggee(CloudDebugger debuggerClient, String uniquifie
 
   @Override
   public DataflowPipelineJob run(Pipeline pipeline) {
+    Set<String> experiments =
```

Comment:
One thing we can do is to always set experiments back to pipeline options when we change something there.